go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy.go

```diff
@@ -32,6 +48,818 @@ func (c *Controller) processNextANPWorkItem(wg *sync.WaitGroup) bool {
+				klog.V(3).Infof("ANP %s's ingress rule %s at priority %d was updated", desiredANPState.name, desiredRule.name, desiredRule.priority)
+				atLeastOneRuleUpdated = true
+			}
+		}
+	}
+
+	// Did ANP.Spec.Egress rules get updated?
+	// (at this stage the length of ANP.Spec.Egress hasn't changed, so individual rules either got updated at their values or positions are switched)
+	// The fields that we care about for rebuilding ACLs are
+	// (i) `ports` (ii) `actions` (iii) priority for a given rule
+	// Did the ANP.Spec.Egress.Peers Change?
+	// 1) ANP.Spec.Egress.Peers.Namespaces changed && ||
+	// 2) ANP.Spec.Egress.Peers.Pods changed && ||
+	// 3) A namespace started or stopped matching the peer && ||
+	// 4) A pod started or stopped matching the peer
+	// If yes we need to recompute the IPs present in our ANP's peer's address-sets
+	if len(desiredANPState.egressRules) == len(currentANPState.egressRules) {
+		for i := range desiredANPState.egressRules {
+			desiredRule := desiredANPState.egressRules[i]
+			currentRule := currentANPState.egressRules[i]
+			ipsToAdd := desiredRule.podIPs.Difference(currentRule.podIPs)
+			klog.Infof("Adding podIPs %+v to address-set %s for ANP %s", ipsToAdd, desiredRule.addrSet.GetName(), desiredANPState.name)
+			addrOps, err := desiredRule.addrSet.AddIPsReturnOps(stringsToIPs(ipsToAdd.UnsortedList()))
+			if err != nil {
+				return fmt.Errorf("unable to construct address-set %s's ip add ops: %v", desiredRule.addrSet.GetName(), err)
+			}
+			ops = append(ops, addrOps...)
+			ipsToRemove := currentRule.podIPs.Difference(desiredRule.podIPs)
+			klog.Infof("Deleting podIPs %+v from address-set %s for ANP %s", ipsToRemove, desiredRule.addrSet.GetName(), desiredANPState.name)
+			addrOps, err = desiredRule.addrSet.DeleteIPsReturnOps(stringsToIPs(ipsToRemove.UnsortedList()))
+			if err != nil {
+				return fmt.Errorf("unable to construct address-set %s's ip delete ops: %v", desiredRule.addrSet.GetName(), err)
+			}
+			ops = append(ops, addrOps...)
+			if currentRule.action != desiredRule.action || !reflect.DeepEqual(desiredRule.ports, currentRule.ports) {
+				klog.V(3).Infof("ANP %s's egress rule %s at priority %d was updated", desiredANPState.name, desiredRule.name, desiredRule.priority)
+				atLeastOneRuleUpdated = true
+			}
+		}
+	}
+	// TODO(tssurya): Check if we can be more efficient by narrowing down exactly which ACL needs a change
+	// The rules which didn't change -> those updates will be no-ops thanks to libovsdb
+	// The rules that changed in terms of their `getACLMutableFields`
+	// will be simply updated since externalIDs will remain the same for these ACLs
+	// No delete ACLs action is required for this scenario
+	// If full aclRecompute=true was done above already, we don't care about individual rule updates...
+	// that will automatically be taken care of in the above transactions
+	if !fullPeerRecompute && atLeastOneRuleUpdated {
```

Comment:
yeah maybe now it makes sense after our latest changes? since earlier i was doing a delete and re-create i think now since we simply use the updateACLsonPG we can do what you said.. let me see if tests pass or if there is some corner case caveat.

Reply:
done, it's good.